### ISSUE TITLE
Hot upgraded app still using old code in custom distillery commands

### ISSUE THREAD

**Issue by dmarkow** [bitwalker/distillery#602]
### Steps to reproduce

* Deploy an app with a custom command
* Perform a hot code upgrade
* Execute the custom command and it will continue to use the old code

### Description of issue

Given a custom command:

```elixir
defmodule Api.ReleaseTasks do
  def foo(_argv) do
    IO.puts Application.spec(:api, :vsn)
  end
end

# rel/config.exs:
# ...
set commands: [
        foo: "rel/commands/foo.sh"
      ]
# ...
```

Before any updates (version 0.1.0)
```
> bin/api foo
"0.1.0"

> bin/api remote_console
Erlang/OTP 21 [erts-10.0.5] [source] [64-bit] [smp:4:4] [ds:4:4:10] [async-threads:1] [hipe]

Interactive Elixir (1.7.2) - press Ctrl+C to exit (type h() ENTER for help)
iex(api@127.0.0.1)1> Api.ReleaseTasks.foo []
"0.1.0"
```

Now I increment my version from 0.1.0 to 0.1.1 and run a distillery upgrade, but the custom command still outputs "0.1.0" from the first release:

```
> bin/api upgrade 0.1.1
...
Made release api:0.1.1 permanent

> bin/api foo
"0.1.0" (this should say 0.1.1?)

> bin/api remote_console
Erlang/OTP 21 [erts-10.0.5] [source] [64-bit] [smp:4:4] [ds:4:4:10] [async-threads:1] [hipe]

Interactive Elixir (1.7.2) - press Ctrl+C to exit (type h() ENTER for help)
iex(api@127.0.0.1)1> Api.ReleaseTasks.foo []
"0.1.1"
```

So while running `remote_console` properly uses the new release code, the custom commands still use the old code. Restarting the running app doesn't help, the only way to get the new custom command code to execute is doing a full (not upgrade) release.

Distillery: 2.0.12, Erlang 1.7.4, Erlang 21.1.4

**Comment by bitwalker** [comment]
Could you paste the contents of `foo.sh`? I'd like to see how you are invoking the command so I can reproduce exactly

**Comment by dmarkow** [comment]
`foo.sh` contents:

```bash
#!/bin/sh

release_ctl eval --mfa "Api.ReleaseTasks.foo/1" --argv -- "$@"
```

**Comment by ecly** [comment]
While I cannot confirm that it is used old code for the ReleaseTask, I can confirm that it is using the wrong version in the executable scripts, meaning that it is looking in the old version's release folder for the migrations.

**Comment by bitwalker** [comment]
This is verified, and addressed in master, the fix has been ported to the 2.0.x branch as well

**Comment by nanlong** [comment]
I have same question. distillery version: 2.1.1
